### PR TITLE
inline tab group flip flag

### PR DESCRIPTION
**Pull request by chassu**
### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

- Flip the flag and remove old tab group related code

### CODE REVIEW

File: Client/Frontend/Browser/Card/Card.swift [hunk 1]
```diff
@@ -245,24 +241,12 @@ struct Card<Details>: View where Details: CardDetails {
 
                 if !details.thumbnailDrawsHeader {
                     HStack(spacing: 0) {
-                        if !FeatureFlag[.tabGroupsNewDesign] && isChildTab(details: details)
-                            && (cardTransitionModel.state == .visibleForTrayShow)
```
Comment (darinwf) [inline Client/Frontend/Browser/Card/Card.swift:249]: I think this means we get to remove the `@EnvironmentObject var cardTransitionModel: CardTransitionModel` line above :-D

File: Client/Frontend/Browser/Card/Card.swift [hunk 2]
```diff
@@ -207,10 +207,6 @@ struct Card<Details>: View where Details: CardDetails {
         GeometryReader { geom in
             VStack(alignment: .center, spacing: 0) {
                 let button = Button(action: {
-                    if let tabGroupCardDetail = tabGroupCardDetail {
-                        tabGroupCardModel.detailedTabGroup = tabGroupCardDetail
```
Comment (darinwf) [inline Client/Frontend/Browser/Card/Card.swift:211]: Does this mean you can get rid of the `@EnvironmentObject var tabGroupCardModel: TabGroupCardModel` above?

**Review by darinwf — APPROVED** [review]
LGTM, please look to see if there are any other `@EnvironmentObject` or `@ObservedObject` lines that are no longer needed. These are free performance wins!

File: Client/Frontend/Browser/Card/CardModels.swift [hunk 3]
```diff
@@ -48,20 +48,17 @@ class TabCardModel: CardModel, TabEventHandler {
...
-        register(self, forTabEvents: .didClose, .didChangeURL)
-        onDataUpdated()
-        self.anyCancellable = manager.objectWillChange.sink { [weak self] (_) in
+        // Process updates to the TabManager state asynchronously, avoiding duplicates. This
+        // way we wait until after `tabs` has been updated before we do any work.
+        self.subscription = manager.$tabs.receive(on: DispatchQueue.main).removeDuplicates().sink {
+            [weak self] (_) in
             if manager.didRestoreAllTabs {
                 self?.onDataUpdated()
                 self?.objectWillChange.send()
```
Comment (evancrow) [inline Client/Frontend/Browser/Card/CardModels.swift:58]: If this is listening for changes, why do you have to call objectWillChange?

**Review by evancrow — APPROVED** [review]
@darinwf Your changes LGTM, with one question

File: Client/Frontend/Browser/Card/CardModels.swift (hunk 3 above)
Comment (darinwf) [reply Client/Frontend/Browser/Card/CardModels.swift:58]: I think that is to signal downstream observers of the TabCardModel... but come to think of it, all of the properties of TabCardModel are `@Published`, so this is probably an unnecessary `objectWillChange.send()`. Hmm...

Comment (darinwf) [reply Client/Frontend/Browser/Card/CardModels.swift:58]: Yes, that `objectWillChange.send()` should be totally redundant with all of the `@Published` properties that get modified by `onDataUpdated`. No need to keep it around.

**Comment by chassu** [comment]
Text fixes LGTM.

**Pull request merged**